.github/workflows/main.yml

```diff
@@ -0,0 +1,51 @@
+name: tt Run and Publish
+
+# Controls when the workflow will run
+on:
+  push:
+    branches: [main]
+  # Allows you to run this workflow manually from the Actions tab
+  workflow_dispatch: 
+    inputs:
+      glossaryDefFile:
+        required: true
+        description: glossaryDefFile
+      deployBranch:
+        required: false
+        description: This is where the output of script will be pushed to.
+
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v2
+      - uses: actions/setup-python@v2.2.2
+      
+      # - name: Install deps
+      #  run: python -m pip install tt
+        
+      - name: Set variables
+        env:
+          DEFAULT_GLOSSARY_DEF_FILE: test
+          DEFAULT_DEPLOY_BRANCH: gh-pages
+        run: |
+          echo "GLOSSARY_DEF_FILE=${{ github.event.inputs.glossaryDefFile || env.DEFAULT_GLOSSARY_DEF_FILE }}" >> $GITHUB_ENV
+          echo "DEPLOY_BRANCH=${{ github.event.inputs.deployBranch || env.DEFAULT_DEPLOY_BRANCH }}" >> $GITHUB_ENV
+
+      - name: Run script
+        run: ./bin/tt glossary ${{ env.GLOSSARY_DEF_FILE }} >index.html
+        
+      - name: Publishing
+        run: |
+          mkdir public
+          cp index.html public
+          cd public
+          git init
```

Comment:
Wouldn't we have to get the current state of the gh-pages branch, and then push an update to that state, instead of just calling git init and starting from scratch each time?